### PR TITLE
Add typescript declarations

### LINKED ISSUE #7: Add TS type declarations
Having type declarations would improve the autocomplete experience for TypeScript users.

### PR DESCRIPTION
**Pull request by TedDriggs**
These improve the autocomplete and error-checking experience. Having them directly in the package 
- rather than distributed through `DefinitelyTyped` - avoids version mismatches and simplifies the install experience for TS users.

Fixes #7

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import { Component, CSSProperties, SFC } from "react";
+
+export interface ScrollbarRendererProps {
+    className?: string | string[];
```
Comment (TedDriggs) [inline index.d.ts]: It looks like this can be tightened to `className: string`, since `join` is always called before the inner renderers are invoked.

Comment (xobotyi) [reply index.d.ts]: But what if i want to pass an array of strings? It is quite popular practice, as far as i know, and i'm using it myself sometimes.  
Anyway i dont see any sense in tightening this one.

**Comment by xobotyi** [comment]
Just to clarify due to TS is unknown for me.
Will this PR cause any extra dependencies or sideeffects for non-ts users?

**Comment by TedDriggs** [comment]
No. For non-TS users, this is no more impactful than `README.md`.

If a new prop is added or the signature of an existing prop is changed, the type declarations will need to be updated for TS users to get a correct autocomplete experience. However, feel free to route those issues to me and I'll push an update, and there won't be any adverse effects before that fix goes in.

**Comment by xobotyi** [comment]
Okay, merging then

**Pull request merged**